daemon/api_snaps.go

```diff
@@ -389,6 +389,14 @@ var errClassicDevmodeConflict = errors.New("cannot use classic and devmode flags
 var errUnaliasedPreferConflict = errors.New("cannot use unaliased and prefer flags together")
 var errNoJailMode = errors.New("this system cannot honour the jailmode flag")
 
+func setDefaultRestartBounderies(st *state.State, tss []*state.TaskSet) error {
+	bootBase, err := snapstate.DeviceModelBase(st)
+	if err != nil {
+		return err
+	}
+	return snapstate.SetEssentialSnapRestartBoundaries(tss, bootBase)
```

Comment:
I find this interface a bit brittle, should snapstate call this for us or is the issue that we have places that use some of its functions and combine the result in larger changes? I see some options

1) Divide the various Install*/Update* functions into some that do this and some that don't ?
2) Maybe 1 is hard, other option is to pass options to those functions not to do this in the few cases it does not matter?
3) some reasonable combination of 1+2
4) keep doing something like that but with a/a set of slightly higher-level function(/s) that produce a Change too at the same time?

I also notice that is a bit unclear that in the next step we can keep doing this the same way for all kind of changes, which would exclude 4 unless it learns about different changes which is a bit strange in some ways

Reply:
I also notice that all the applications of this would need to be tested

Reply:
I think your suggestions here are spot-on. I think your second suggestion combined with some default behaviour in snapstate may be the right way to go. I'd like to just experiment a bit with that.

Reply:
I've updated the PR. This is indeed much better than the previous approach. Does the new approach make better sense?